### PR TITLE
feat: april updates

### PR DESCRIPTION
**Pull request by Julien-R44**

### CODE REVIEW

File: content/blog_posts/articles/april-2024-updates.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+- Documentation: https://adonisjs.com/blog/hmr-in-adonisjs
+
+- Article: https://docs.adonisjs.com/guides/hmr
+
+## Documentation improvements
+
+Some weeks ago, we opened [an issue](https://github.com/adonisjs/v6-docs/issues/80) on the documentation repo to discuss restructuring the docs. Many of you read and shared your ideas. We've incorporated your feedback and have started working on a new structure, that is now [available and published](https://docs.adonisjs.com/). 
+
+![alt text](new_docs.png)
```
Comment (RomainLanz) [inline content/blog_posts/articles/april-2024-updates.md:21]:
```suggestion
![](new_docs.png)
```

File: content/blog_posts/articles/april-2024-updates.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+- Documentation: https://docs.adonisjs.com/guides/references/commands#envadd
+
+## InferPageProps in Inertia
+
+We've shipped an utility type in `@adonisjs/inertia` that makes it easier to share types between your front-end and back-end. 
+
+```ts
+export default class UsersController {
+  async index({ inertia }: HttpContextContract) {
```
Comment (RomainLanz) [inline content/blog_posts/articles/april-2024-updates.md:47]:
```suggestion
  async index({ inertia }: HttpContext) {
```

File: content/blog_posts/articles/april-2024-updates.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+      { id: 1, name: 'John Doe', createdAt: new Date() }
+    ]
+
+    return inertia.render('users/index', { users })
+  }
+}
+```
+
+```ts
+import { InferPageProps } from '@adonisjs/inertia'
```
Comment (RomainLanz) [inline content/blog_posts/articles/april-2024-updates.md:58]:
```suggestion
import type { InferPageProps } from '@adonisjs/inertia'
```

File: content/blog_posts/articles/april-2024-updates.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+function UsersList(
+  props: InferPageProps<UsersController, 'index'>
+) {
+  props.users
+  //      ^? { id: number, name: string, createdAt: string }[]
+}
+```
+
+This feature should help save time and reduce typing errors in your Inertia applications. Note that the `createdAt` field is automatically converted to a string in your front-end component type, as Dates are serialized into strings when passed over the wire in JSON. This `InferPageProps` type performs this serialization at the type level automatically for you.
```
Comment (RomainLanz) [inline content/blog_posts/articles/april-2024-updates.md:69]: Should we talk about the serialization of `ModelObject` here? I suggest adding a small note that points to a section in the documentation.

**Review by RomainLanz — COMMENTED** [review]
I believe we can also add Transmit and state that an article is coming in the following days.

File: content/blog_posts/articles/april-2024-updates.md (hunk 4 above)
Comment (Julien-R44) [reply content/blog_posts/articles/april-2024-updates.md:69]: I think we can keep it, to me this is a cool feature to showcase

**Comment by Julien-R44** [comment]
Added transmit

**Comment by thetutlage** [comment]
Let's mention the following release in "Incase you missed" section. Rest all looks good.

https://github.com/adonisjs/lucid/releases/tag/v20.6.0

**Pull request merged**